pkg/download/jsoncreator/json_creator.go

```diff
@@ -131,6 +132,33 @@ func replaceKeyProperties(dat map[string]interface{}) map[string]interface{} {
 	return dat
 }
 
+// removes key with specified path
+func removeKey(dat map[string]interface{}, key []string) map[string]interface{} {
+	if len(key) == 0 {
+		return dat
+	} else if len(key) == 1 {
```

Comment:
as you fail early here and return out of your ifs you can skip the else and improve readability

```go
if len(key) == 0 {
    return dat
}

if len(key) == 1 {
    delete(dat, key[0])
    return dat
}

[the logic in the normal case]
```